Create redesigned "Users" page

vorleakyek opened the pull request:
Fixes #1751 

### What changes did you make and why did you make them ?

  - Create redesigned "Users" page `/users/` with links to `/users/user-search` and `/users/permission-search` as separate buttons. 
  - Click on "Users" takes the user to the new, redesigned "Users" page
  - Change "User Management" title on existing User Management page to "User Search"
  - Note: There is a follow up ticket to handle the user permission search page in #1752. So, clicking on the button might not work as expected yet.  

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

![users-page-previous](https://github.com/user-attachments/assets/403fd5e3-098e-4c55-afaa-4f6408ff05f2)


</details>

<details>
<summary>Visuals after changes are applied</summary>
  
![redesigned-users-page](https://github.com/user-attachments/assets/520f9bcc-c214-4236-902a-6b56bd2e4fd5)


</details>

vorleakyek committed:
Add redesigned Users page

vorleakyek committed:
Create redesigned Users page with two buttons link to users-search and users-permission-search

pluto-bell reviewed (commented):
Great work! MUI is being used for the buttons, the styling looks correct and the links work as expected (user permission search page to be added).

I noticed that the `package.json`, `package-lock.json` and `yarn.lock` files show updates, I don't know enough about back-end to say whether this would be an issue to merge or not, so I will leave this to the lead devs.

Edit: I'd love to hear from a lead dev on the above comment so I can know how to approach moving forward.

vorleakyek committed:
undo deletion of package.json and yarn.lock

vorleakyek commented:
@pluto-bell, thanks for taking a look at the PR. I think you're right that the package.json, package-lock.json and yarn.lock files shouldn't be included in the PR. 

@trillium, is this okay? I tried to remove those files from the PR, but it's still showing this in the files changed. If not, what is the proper command to remove the files from PR?

![image](https://github.com/user-attachments/assets/c0a6165f-185a-478b-b865-f9a1b77908a3)

JackHaeg commented:
Per @trillium: restore the files using:

`git checkout development -- FILEPATH1 FILEPATH2 `

@trillium will plan to meet with @vorleakyek this week to restore the json files.

vorleakyek committed:
Update div to box

vorleakyek commented:
Thanks @trillium for helping me restore the JSON files yesterday. 
I also updated the code in `client/src/pages/Users.jsx` to use "Box" and "Container" MUI components instead of the "div" element,  and updated the style in `client/src/sass/Users.scss` file.

pluto-bell reviewed (approved):
Great, everything looks good to me and is working as expected (:

trillium commented:
PR has been added to [user_permissions_search](https://github.com/hackforla/VRMS/tree/user_permissions_search) feature branch!
